src/main.rs

```diff
@@ -1,3 +1,89 @@
-fn main() {
-    println!("Hello, World!");
+use std::fs;
+use anyhow::{Context, Result};
+use walkdir::WalkDir;
+
+mod arg;
+
+fn main() -> Result<()> {
+    let args = arg::parse_args();
+
+    println!("Cleaning directory: {:?}", args.dir);
+    println!("File types to clean: {:?}", args.types);
+    println!("Minimum file size: {} bytes", args.min_size);
+
+    directory_cleaner(&args.dir, &args.types, args.min_size)?;
+
+    println!("Cleaning completed successfully.");
+
+    Ok(())
 }
+
+
+fn directory_cleaner(dir: &String, paths_to_clear: &[String], size: u64) -> Result<()> {
+    // skip files that cannot be opened, due to permission issues etc
+    for entry in WalkDir::new(dir).into_iter().filter_map(|f| f.ok()) {
+        let path = entry.path();
+
+        if path.is_file() {
+            let ext = path.extension()
+                .and_then(|ex| ex.to_str())
+                .unwrap_or("");
+
+            if paths_to_clear.iter().any(|p| ext == p) {
+                let metadata = fs::metadata(path)
+                    .with_context(|| format!("Failed to read metadata for file: {:?}", path))?;
+
+                if metadata.len() >= size {
```

Comment:
why are we doing this check?

Reply:
To delete the files with size above the min_size provided by the user (defaults to O bytes if non is given)

Reply:
not part of the task ):